core/src/main/java/org/geysermc/geyser/session/GeyserSession.java

```diff
@@ -2044,6 +2146,27 @@ public void removeFog(String... fogNameSpaces) {
         return Set.copyOf(this.appliedFog);
     }
 
+    @Override
+    public void lockInputs(boolean camera, boolean movement) {
```

Comment:
I wonder if, for something like this, we should have something like what Bukkit occasionally does where we track the locks based on each extension that wants it? Or we should prepare for such an idea if it comes up.
https://jd.papermc.io/paper/1.20/org/bukkit/entity/Player.html#showEntity(org.bukkit.plugin.Plugin,org.bukkit.entity.Entity)

Reply:
Not a bad idea, yeah. Now that I think of it, we should also probably expose the current locks of a connection too.

One issue this approach would pose though - not just extensions can call it, so we would need some other way to track who did it.

Reply:
what if lockInputs returned a new random UUID

Reply:
In other words, the UUID would serve as a key to unlock the inputs again?

I could see that being fairly confusing tbh

Reply:
yes - owner ID

i don't know what to say about possibility of confusion - if we doc it well then i'm okay with it. having an `Owner` class or something seems excessive considering it would just be an ID wrapper.

unless we have extensions implement Owner, and provide a default implementation that hashes the Extension name. which seems sensible.

the event system already has some sort of thing like this (extension vs outsider)

Reply:
Implemented and separated them into lockCamera and lockMovement